src/main/java/com/phamhieu/bookapi/domain/user/UserError.java

```diff
@@ -0,0 +1,28 @@
+package com.phamhieu.bookapi.domain.user;
+
+import com.phamhieu.bookapi.error.BadRequestException;
+import com.phamhieu.bookapi.error.NotFoundException;
+import lombok.experimental.UtilityClass;
+
+import java.util.UUID;
+import java.util.function.Supplier;
+
+@UtilityClass
+public class UserError {
+
+    public static Supplier<NotFoundException> supplyUserNotFoundById(final UUID id) {
+        return () -> new NotFoundException("User has id:  %s couldn't be found", id);
+    }
+
+    public static Supplier<NotFoundException> supplyUserNotFoundByName(final String name) {
+        return () -> new NotFoundException("User has name:  %s couldn't be found", name);
```

Comment:
Mình không dùng tên để phân biệt user nên không có cái này

```suggestion
        return () -> new NotFoundException("User has name: %s couldn't be found", name);
```